platform/vcs-impl/src/com/intellij/openapi/vcs/update/AbstractCommonUpdateAction.java

```diff
@@ -301,7 +301,7 @@ private static boolean supportingVcsesAreEmpty(final ProjectLevelVcsManager vcsM
     private LocalHistoryAction myLocalHistoryAction;
 
     public Updater(final Project project, final FilePath[] roots, final Map<AbstractVcs, Collection<FilePath>> vcsToVirtualFiles) {
-      super(project, getTemplatePresentation().getText(), true, VcsConfiguration.getInstance(project).getUpdateOption());
+      super(project, getTemplatePresentation().getText(), true, ActionInfoEx.maybeOverrideBackGroundOption(myActionInfo, VcsConfiguration.getInstance(project).getUpdateOption()));
```

Comment:
Could you please shed some light on what update events should happen modally? I'm asking because we were in fact almost ready to remove this setting at all, suspecting (according to our personal feelings and the statistics we get) that nobody updates the project modally.